### PR TITLE
Added new DAG for JAX Stable Stack Maxtext End to End Tests

### PR DESCRIPTION
**Pull request by parambole**
# Description

- Scaffolding of JAX Stable stack config 
- Added Maxtext tests for V4 and V5P single and multi slice configuration 
# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**List links for your tests (use go/shortn-gen for any internal link):** ...
https://c3aed121fd4247389a3ca4b9c5878779-dot-us-east4.composer.googleusercontent.com/dags/maxtext_jax_ss_e2e/grid

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ X] I have performed a self-review of my code.
- [ X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ X] I have run one-shot tests and provided workload links above if applicable. 
- [ X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/vm_resource.py [hunk 1]
```diff
@@ -165,6 +165,9 @@ class DockerImage(enum.Enum):
       "gcr.io/tpu-prod-env-multipod/maxtext_jax_stable:"
       f"{datetime.datetime.today().strftime('%Y-%m-%d')}"
   )
+  MAXTEXT_TPU_JAX_SS = (
```
Comment (RissyRan) [inline dags/vm_resource.py:168]: Would like to double check if we expect one pinned version for now. Thanks!

File: dags/imagegen_devx/maxtext_jax_ss_e2e.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+# Run once a day at 4 am UTC (8 pm PST)
+SCHEDULED_TIME = "0 4 * * *" if composer_env.is_prod_env() else None
+
+
+with models.DAG(
+    dag_id="maxtext_jax_ss_e2e",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "jax-ss"],
+    start_date=datetime.datetime(2024, 1, 19),
```
Comment (RissyRan) [inline dags/imagegen_devx/maxtext_jax_ss_e2e.py:33]: nit: probably update to current date?

**Review by RissyRan — COMMENTED** [review]
Thanks! LGTM! Do you want to add you as [test owner](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/.github/CODEOWNERS#L22-L23) of jax ss dag?

File: dags/vm_resource.py (hunk 1 above)
Comment (parambole) [reply dags/vm_resource.py:168]: Yes, that is correct.

File: dags/imagegen_devx/maxtext_jax_ss_e2e.py (hunk 2 above)
Comment (parambole) [reply dags/imagegen_devx/maxtext_jax_ss_e2e.py:33]: Done

**Pull request merged**